metacat-metadata-mysql/src/main/java/com/netflix/metacat/metadata/mysql/MySqlParentChildRelMetaDataService.java

```diff
@@ -0,0 +1,182 @@
+package com.netflix.metacat.metadata.mysql;
+
+import com.netflix.metacat.common.QualifiedName;
+import com.netflix.metacat.common.server.model.ChildInfo;
+import com.netflix.metacat.common.server.model.ParentInfo;
+import com.netflix.metacat.common.server.usermetadata.ParentChildRelMetadataService;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.retry.RetryCallback;
+import org.springframework.retry.support.RetryTemplate;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.sql.PreparedStatement;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Parent Child Relationship Metadata Service.
+ * This stores the parent child relationship of two entities as first class citizen in Metacat.
+ */
+@Slf4j
+@SuppressFBWarnings
+@Transactional("metadataTxManager")
+public class MySqlParentChildRelMetaDataService implements ParentChildRelMetadataService {
+    // For now,
+    // First where clause ? = parent : Prevent the table that is a child for another to be created as a parent
+    // Second where clause ? = child : Prevent the child having more than one parent
+    static final String SQL_CREATE_PARENT_CHILD_RELATIONS =
```

Comment:
For read committed level, which can still have phatom read, we can still have two concurrent transaction pass the exist call and end up with invalid result

Given the limitation, probably we can separate the validation out of this SQL?